scrapper.py

```diff
@@ -25,17 +39,36 @@ class Crawler:
     """
     Crawler implementation
     """
-    def __init__(self, seed_urls, max_articles: int):
-        pass
+
+    def __init__(self, seed_urls, total_max_articles):
+        self.seed_urls = seed_urls
+        self.total_max_articles = total_max_articles
+        self.urls = []
 
     def _extract_url(self, article_bs):
-        pass
+        urls_bs = article_bs.find_all('div', class_='title')
+        fulls = []
+        for url_bs in urls_bs:
+            part_of_url = url_bs.find('a')['href']
+            if 'entry' in part_of_url:
+                fulls.append(HTTP_PATTERN + part_of_url)
+
+        for full in fulls:
+            if len(self.urls) < self.total_max_articles:
+                self.urls.append(full)
```

Comment:
Before adding a URL to self.urls, it is recommended to check that you have not yet saved it before as in many sources links repeat themselves multiple times, and we only want  to collect unique ones

Reply:
now it checks